src/1_utils.js

```diff
@@ -105,20 +105,66 @@ utils.whiteListSessionData = function(data) {
 	};
 };
 
+/**
+ * Abstract away the window.location for better testing
+ */
+utils.getWindowLocation = function() {
+	return String(window.location);
+};
+
 utils.cleanLinkData = function(linkData) {
 	linkData['source'] = 'web-sdk';
-	if (linkData['data'] && linkData['data']['$desktop_url'] !== undefined) {
```

Comment:
The check for desktop_url used to be the only one, but once we started adding more types, I went for a refactor that checked desktop_url on line 152 of the newer code.  And then, in that case, I am specifically looking for a 'string' type, instead of the absence of the value.